src/api/app/controllers/webui/request_controller.rb

```diff
@@ -352,9 +352,11 @@ def addreview_opts
   end
 
   def any_project_maintained_by_current_user?
-    projects = @actions.select(:target_project).distinct.pluck(:target_project)
+    projects = Project.where(name: @actions.select(:target_project)).distinct
     maintainer_role = Role.find_by_title('maintainer')
-    projects.any? { |project| Project.find_by_name(project).user_has_role?(User.possibly_nobody, maintainer_role) }
+    user = User.possibly_nobody
+
+    projects.any? { |project| project.user_has_role?(user, maintainer_role) }
```

Comment:
Sounds to me like this could be a one liner:

`Project.for_user(User.possibly_nobody).where(name: @actions.select(:target_project)).any?`

Reply:
Yeah... I thought about some one-liner refactoring using `Relationship` instead of `Project`... Then I realized `user_has_role?` also checks for permissions of the groups a user belongs to. See: https://github.com/openSUSE/open-build-service/blob/3c92fcda6b248dc8ad059476c81673fd76c60955/src/api/app/models/concerns/has_relationships.rb#L42-L46

That option could work if we stop taking in account maintainer roles given to groups...

Reply:
There is also `Project.for_groups`, you can combine those.

Reply:
Sure. But that is not the purpose of the pull request. This PR fixes the issue. We can leave the possible refactoring for later on.

Reply:
You are already significantly refactoring the code in this PR.

Reply:
What about this alternative?

```ruby
Relationship.maintainers.where(project: projects, user: user).or(Relationship.where(group: [user.groups.unscope(:order)])).any?
```

A trigger of this query in my development environment returned:
```
Relationship Exists? (0.6ms)  SELECT 1 AS one FROM `relationships` WHERE (
  `relationships`.`role_id` = 2 AND
  `relationships`.`project_id` IN (
    SELECT `projects`.`id` FROM `projects` WHERE `projects`.`id` != 0 AND `projects`.`id` = 7
  ) AND
  `relationships`.`user_id` = 4 OR
  `relationships`.`group_id` IN (
    SELECT DISTINCT `groups`.`id` FROM `groups` INNER JOIN `groups_users` ON `groups`.`id` = `groups_users`.`group_id` WHERE `groups_users`.`user_id` = 4
  )
) LIMIT 1
```